R/survival_reg-data.R

```diff
@@ -296,4 +296,166 @@ make_survival_reg_flexsurv <- function() {
   )
 }
 
+make_survival_reg_flexsurvspline <- function() {
+
+  parsnip::set_model_engine("survival_reg",
+                            mode = "censored regression",
+                            eng = "flexsurvspline")
+  parsnip::set_dependency("survival_reg",
+                          eng = "flexsurvspline",
+                          pkg = "flexsurv",
+                          mode = "censored regression")
+  parsnip::set_dependency("survival_reg",
+                          eng = "flexsurvspline",
+                          pkg = "survival",
+                          mode = "censored regression")
+  parsnip::set_dependency("survival_reg",
+                          eng = "flexsurvspline",
+                          pkg = "censored",
+                          mode = "censored regression")
+
+  parsnip::set_model_arg(
+    model = "survival_reg",
+    eng = "flexsurvspline",
+    parsnip = "num_knots",
+    original = "k",
+    func = list(pkg = "dials", fun = "num_knots"),
+    has_submodel = FALSE
+  )
+
+  parsnip::set_model_arg(
+    model = "survival_reg",
+    eng = "flexsurvspline",
+    parsnip = "survival_link",
+    original = "scale",
+    func = list(pkg = "dials", fun = "survival_link"),
+    has_submodel = FALSE
+  )
```

Comment:
Full disclosure: this still won't work down the line with the tune package atm because tuning does not work yet for any models in the `"censored regression"` mode - but this will open that up, once we touch the tune package.

Reply:
`num_knots` should be tunable. Not sure it is reasonable to tune the survival link function, so maybe that can just be set as an engine specific argument without tuning.

Reply:
If you don't want the link function to be a tunable arg (which I could understand from a stats standpoint), then we don't have to do anything else about it, it can just be set as `scale` in `set_engine()`.